main.nf

```diff
@@ -425,19 +462,30 @@ process edgeR_miRBase_mature {
     dataDGE <- dataDGE[o,]
     dataNorm <- calcNormFactors(dataDGE)
 
-    # Make MDS plot
-    pdf(paste(header,"_edgeR_MDS_plot.pdf",sep=""))
-    MDSdata <- plotMDS(dataNorm)
+    # Print normalized read counts to file
+    dataNorm_df<-as.data.frame(cpm(dataNorm))
+    write.table(dataNorm_df,file=paste(header,"_normalized_CPM.txt",sep=""),sep='\\t',quote=FALSE)
+
+    # Print heatmap based on normalized read counts
+    pdf(paste(header,"_CPM_heatmap.pdf",sep=""))
+    heatmap.2(cpm(dataNorm),col=redgreen(100),key=TRUE,scale="row",density.info="none",trace="none")
     dev.off()
 
+    # Make MDS plot (only perform with 3 or more samples)
+    if (ncol(data)>2){
+      pdf(paste(header,"_edgeR_MDS_plot.pdf",sep=""))
+      MDSdata <- plotMDS(dataNorm)
+      dev.off()
+    }
+
     # Print distance matrix to file
-    write.table(MDSdata\$distance.matrix, paste(header,"_edgeR_MDS_distance_matrix.txt",sep=""), quote=FALSE, sep="\\t")
+    # write.table(MDSdata\$distance.matrix, paste(header,"_edgeR_MDS_distance_matrix.txt",sep=""), quote=FALSE, sep="\\t")
```

Comment:
I forgot to ask.. Why are you commenting out / deleting these files? The heatmap and MDS plot in the MultiQC report will disappear without them..